der/src/asn1.rs

```diff
@@ -43,7 +43,7 @@ pub use self::{
     sequence_of::{SequenceOf, SequenceOfIter},
     set_of::{SetOf, SetOfIter},
     teletex_string::TeletexStringRef,
-    utc_time::UtcTime,
+    utc_time::{UtcTime, MAX_YEAR},
```

Comment:
This needs to be split and released to fix  `minimal-versions` test

Reply:
Importing that into the `asn1` module doesn't make much sense because it's specific to `UtcTime`. `GeneralizedTime` has a different max year, for example.

How about making it an inherent constant?

Reply:
what do you mean by that? The only purpose was to make a conversion between `GeneralTime` and `UTCTime` to comply with https://www.rfc-editor.org/rfc/rfc5280#section-4.1.2.5:
```
   CAs conforming to this profile MUST always encode certificate
   validity dates through the year 2049 as UTCTime; certificate validity
   dates in 2050 or later MUST be encoded as GeneralizedTime.
   Conforming applications MUST be able to process validity dates that
   are encoded in either UTCTime or GeneralizedTime.
```

Reply:
The constant would be named `der::asn1::MAX_YEAR`, but that isn't the maximum year expressible by "ASN.1", it's the maximum year expressible by `UTCTime`

Reply:
Ha, yes that makes sense.